src/3d/terrain/qgsdemterraintileloader_p.cpp

```diff
@@ -52,7 +52,7 @@ static void _heightMapMinMax( const QByteArray &heightMap, float &zMin, float &z
 }
 
 
-QgsDemTerrainTileLoader::QgsDemTerrainTileLoader( QgsTerrainEntity *terrain, QgsChunkNode *node )
+QgsDemTerrainTileLoader::QgsDemTerrainTileLoader( QgsTerrainEntity *terrain, QgsChunkNode *node, bool loadSynchronously )
```

Comment:
Hmm I would like to avoid having a new `loadSynchornously` flag in terrain tile loaders. I would suggest that instead we have a virtual method `waitForFinished()` that would block until the loading has not finished. This should keep the code cleaner and easier to test because there will be fewer possible code paths.